tristan/ding-wall/src/components/ListTasks.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import './ListTasks.css';
+import ListItem from './ListItem';
+import InputForm from './InputForm';
+
+/* I'm having trouble organising this app in a way that conforms to
+my understanding of good, component-based, React programming.
+
+E.g.: I don't like the "const xxYy = zzz => {}" syntax. I much prefer the 
+eqivalent(?) functional syntax (function xxYy(zzz) = {}), though it seems that 
+functions written in this way ought to be saved as components in a separate file. In the 
+case of the below, the addTask function makes use of the setTask state setting function,
+so (presumably) cannot be saved as a separate file. This means that my InputForm 
+component also cannot be a separate file, since it makes use of addTask.
+
+It makes sense to me that there should be separate components 
+responsible for rendering the input form (InputForm) and the list of tasks
+(ListTasks) -- see https://reactjs.org/docs/thinking-in-react.html --
+yet because of the reasons above, it seems necessary to have a single file that 
+contains both compnents. Need some guidance on this one... 
+
+EDIT: I figured most of this out, but I'm leaving the comment for 
+future discussion.
```

Comment:
It seems like you've figured out the concept that while the component code lives in another file, you import it and use it here. I think something that will help make this a bit more clear is trying to remember that **a component is just a function**. 

The concept of importing and reusing functions throughout code is by no means new, and that's all that's going on here. The JSX syntax makes it look like html, but in fact you are just calling the function, and passing it parameters.

i.e.
```jsx
// these two lines mean the same thing
<InputForm addTask={addTask} />

React.createElement(
  InputForm,
  {addTask: addTask},
  null
)
```

To your original point, using the arrow function syntax ```const component = props => {...}``` or the function syntax ```function component(props) {...}``` is a choice you can make. Many people prefer the function syntax for a number of valid reasons.

There is a difference between the two which is really worth exploring and understanding at this point. I'd like to have a video call with you and work through some stuff to get a solid grasp on the difference. Essentially, there are two major differences.

1. The functional syntax `hoists` the function definition, so it can be used **before it's been declared** because the JavaScript parser runs through the file and searches for these before it runs through and starts to execute the code.
2. The `context` of the function differs. This is a bit harder to grasp at first. In JavaScript, and many programming languages, the concept of `this` is used to access properties of the current object or scope (sometimes it's `self`, or even `me`). In function syntax, `this` is relative to the function's scope, so anything outside the function will not be available unless we modify the scope using [bind](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Function/bind), [apply](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply), or [call](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call). In arrow functions, `this` is implicitly bound to the context in which the function is declared. 

Again, these are not obvious distinctions, and it takes some time working with them to really get a feel of what each syntax can do for us as developers. Both are useful. If you prefer function, that's fine, use it. I like it too.

Regarding the point of declaring the addTask function in a separate file, it is possible, but I think it would be a more complicated abstraction. In React we keep the business logic and UI together, which is not commonly done. This is just something to get used to.